metricbeat/module/mongodb/status/status.go

```diff
@@ -17,43 +15,52 @@ TODOs:
 	* add a metricset for "metrics" data
 */
 
-var debugf = logp.MakeDebug("mongodb.status")
-
 func init() {
 	if err := mb.Registry.AddMetricSet("mongodb", "status", New, mongodb.ParseURL); err != nil {
 		panic(err)
 	}
 }
 
+// MetricSet type defines all fields of the MetricSet
+// As a minimum it must inherit the mb.BaseMetricSet fields, but can be extended with
+// additional entries. These variables can be used to persist data or configuration between
+// multiple fetch calls.
 type MetricSet struct {
 	mb.BaseMetricSet
-	dialInfo *mgo.DialInfo
+	dialInfo     *mgo.DialInfo
+	mongoSession *mgo.Session
 }
 
+// New creates a new instance of the MetricSet
+// Part of new is also setting up the configuration by processing additional
+// configuration entries if needed.
 func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 	dialInfo, err := mgo.ParseURL(base.HostData().URI)
 	if err != nil {
 		return nil, err
 	}
 	dialInfo.Timeout = base.Module().Config().Timeout
 
+	// instantiate direct connections to Mongo host
+	mongoSession, err := mongodb.NewDirectSession(dialInfo)
+	if err != nil {
+		return nil, err
+	}
+
 	return &MetricSet{
 		BaseMetricSet: base,
 		dialInfo:      dialInfo,
+		mongoSession:  mongoSession,
 	}, nil
 }
 
+// Fetch methods implements the data gathering and data conversion to the right format
+// It returns the event which is then forward to the output. In case of an error, a
+// descriptive error must be returned.
 func (m *MetricSet) Fetch() (common.MapStr, error) {
-	session, err := mgo.DialWithInfo(m.dialInfo)
-	if err != nil {
-		return nil, err
-	}
-	defer session.Close()
-
-	session.SetMode(mgo.Monotonic, true)
```

Comment:
@ruflin 

Yes, this was intentional. [Monotonic Mode](https://godoc.org/gopkg.in/mgo.v2#Mode) connects to the cluster as a whole and reads from the nearest secondary (not clear how nearest is defined). So, if your desired target is busy or goes down, mgo will read from a  different node in the cluster. This failover strategy is what I was referring to earlier (which I think is undesirable) - where you may read from a node other than the target specified in the module's configuration, and obfuscate node failures. 

Thus, the change I've implemented establishes [direct connections](https://godoc.org/gopkg.in/mgo.v2#DialInfo.Direct) to each of the nodes specified in the module's settings. You can see how I did it [here](https://github.com/scottcrespo/beats/blob/mongo-dbstats/metricbeat/module/mongodb/mongodb.go#L71). This ensures that we are reading from the intended node. And, if it's slow or fails, we will know about it. 

Also, note that I've also set [FailFast](https://godoc.org/gopkg.in/mgo.v2#DialInfo.FailFast) to `True`, which helps identity latent and non-responsive servers sooner rather than later.